### PR TITLE
Add tabular stormlib (SYN-7652)

### PR DESCRIPTION
**Pull request by mikemoritz**

### CODE REVIEW

File: synapse/lib/stormlib/tabular.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    _storm_locals = (
+        {'name': 'row', 'desc': 'Create a new row string from a data list.',
+         'type': {'type': 'function', '_funcname': 'row',
+                  'args': (
+                      {'name': 'data', 'type': 'list',
+                       'desc': 'The data to create the row from; length must match the number of configured columns.'},
+                  ),
+                  'returns': {'type': 'str', 'desc': 'The row string.'}}},
+        {'name': 'header',
+         'desc': 'Create a heaer row string.',
```
Comment (vEpiphyte) [inline synapse/lib/stormlib/tabular.py:115]: header

File: synapse/lib/stormlib/cortex.py [hunk 2]
```diff
@@ -20,18 +20,29 @@
         'storm': '''
         $apis = $lib.cortex.httpapi.list()
         if $apis {
-            $header = 'order iden                             owner                auth  runas  path'
-            $lib.print($header)
+            $config = (({
```
Comment (vEpiphyte) [inline synapse/lib/stormlib/cortex.py:23]: Noodling this - we could put this type of configuration data into `cmdconf` and avoid having to parse it in storm.

File: synapse/lib/stormlib/tabular.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    async def _methPrinter(self, conf):
+        conf = await s_stormtypes.toprim(conf)
+        conf.setdefault('separators', {})
+        conf = s_schemas.reqValidTabularConf(conf)
+        return TabularPrinter(self.runt, conf)
+
+@s_stormtypes.registry.registerType
+class TabularPrinter(s_stormtypes.StormType):
+    '''
+    A storm object for printing tabular data using a defined configuration.
```
Comment (Cisphyx) [inline synapse/lib/stormlib/tabular.py:103]:
```suggestion
    A Storm object for printing tabular data using a defined configuration.
```

**Review by Cisphyx — DISMISSED** [review]

File: synapse/lib/schemas.py [hunk 4]
```diff
@@ -318,3 +318,52 @@
...
+                             'description': 'The column name which will be used in the header row.'},
+                    'width': {'type': 'number', 'default': None, 'exclusiveMinimum': 0,
+                              'description': 'If not provided each cell will expand to fit the data.'},
+                    'justify': {'type': 'string', 'default': 'left', 'enum': ['left', 'center', 'right'],
+                                'description': 'Justification for the header titles and data rows.'},
+                    'overflow': {'type': 'string', 'default': 'trim', 'enum': ['wrap', 'trim'],
+                                 'description': 'For text exceeding the width, '
+                                                'either wrap text in multiple lines or trim and append "...".'},
+                    'newlines': {'type': 'string', 'default': 'replace', 'enum': ['replace', 'split'],
+                                'description': 'Replace newlines with a space or split into multiple lines.'
```
Comment (Cisphyx) [inline synapse/lib/schemas.py:356]:
```suggestion
                                 'description': 'Replace newlines with a space or split into multiple lines.'
```

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**